Framework/PythonInterface/plugins/algorithms/SCDAdjustSampleNorm.py

```diff
@@ -0,0 +1,246 @@
+        self.setPropertySettings("BinningDim1", histo_settings)
+        self.setPropertySettings("BinningDim2", histo_settings)
+
+        self.declareProperty(
+            WorkspaceProperty("OutputWorkspace", "", optional=PropertyMode.Mandatory, direction=Direction.Output),
+            doc="Output MDWorkspace in Q-space, name is prefix if multiple input files were provided.")
+
+    def validateInputs(self):
+        issues = dict()
+
+        filelist = self.getProperty("Filename").value
+        vanfile = self.getProperty("VanadiumFile").value
+        input_ws = self.getProperty("InputWorkspaces")
+        van_ws = self.getProperty("VanadiumWorkspace")
+
+        # Make sure files and workspaces aren't both set
+        if len(filelist) >= 1:
+            if not input_ws.isDefault:
+                issues['InputWorkspaces'] = "Cannot specify both a filename and input workspace"
+        else:
+            if input_ws.isDefault:
+                issues['Filename'] = "Either a file or input workspace must be specified"
+
+        if len(vanfile) <= 0 and van_ws.isDefault:
+            issues['VanadiumFile'] = "Either a vanadium file or vanadium workspace must be specified!"
+
+        if len(vanfile) > 0 and not van_ws.isDefault:
+            issues['VanadiumWorkspace'] = "Cannot specify both a vanadium file and workspace"
+
+        # Verify given workspaces exist
+        if not input_ws.isDefault:
+            input_ws_list = input_ws.value.split(",")
+            for ws in input_ws_list:
+                if not mtd.doesExist(ws):
+                    issues['InputWorkspaces'] = "Could not find input workspace '{}'".format(ws)
+
+        return issues
+
+    def PyExec(self):
+        load_van = not self.getProperty("VanadiumFile").isDefault
+        load_files = not self.getProperty("Filename").isDefault
+
+        if load_files:
+            datafiles = self.getProperty("Filename").value
+        else:
+            datafiles = self.getProperty("InputWorkspaces").value.split(",")
+
+        vanadiumfile = self.getProperty("VanadiumFile").value
+        vanws = self.getProperty("VanadiumWorkspace").value
+        height = self.getProperty("DetectorHeightOffset").value
+        distance = self.getProperty("DetectorDistanceOffset").value
+        method = self.getProperty("OutputAsMDEventWorkspace").value
+
+        if method:
+            minvals = self.getProperty("MinValues").value
+            maxvals = self.getProperty("MaxValues").value
+            merge = self.getProperty("MergeInputs").value
+            wslist = []
+        else:
+            bin0 = self.getProperty("BinningDim0").value
+            bin1 = self.getProperty("BinningDim1").value
+            bin2 = self.getProperty("BinningDim2").value
+
+        out_ws = self.getPropertyValue("OutputWorkspace")
+        out_ws_name = out_ws
+
+        if load_van:
+            vanws = Load(vanadiumfile, StoreInADS=False)
```

Comment:
You can use LoadMD instead of Load, this will reduced the overhead from Load from when it determines the file type and which loader to use.